packages/qwik-labs/src/insights/index.tsx

```diff
@@ -164,13 +170,26 @@ function symbolTracker(
     sessionID,
   };
   let timeoutID: ReturnType<typeof setTimeout> | null;
+  let qRouteChangeTime = new Date().getTime();
+  const qRouteEl = document.querySelector('[q\\:route]');
+  if (qRouteEl) {
+    const observer = new MutationObserver((mutations) => {
+      const mutation = mutations.find((m) => m.attributeName === 'q:route');
+      if (mutation) {
+        qRouteChangeTime = new Date().getTime();
+      }
+    });
+    observer.observe(qRouteEl, { attributes: true });
```

Comment:
So this logic is missing something. When `q:route` changes we must flush the data, because the `route` is shared on `payload` and if we don't flush we will attribute the symbols to the wrong route. 

So a `q:route` change must flush the current symbols and start a new queue.

Reply:
I will add the route in the symbol so we can avoid extra logic

Reply:
Fixed with [2ed9ae3](https://github.com/BuilderIO/qwik/pull/4971/commits/2ed9ae3f265344979d96200848c36e7c98ae7043)